### PR TITLE
Sync assemblies

### LINKED ISSUE #123: Sychronize assembled genomes along with sequence files when synchronizing samples
Imported from GitLab issue #563.  Originally posted on 2017/11/10 10:56AM

### PR DESCRIPTION
**Pull request by tom114**
## Description of changes
Added the ability to synchornize remote GenomeAssemblies when synchronizing a project.

To test this PR, upload some assemblies to a sample in a project, they synchronize it using the sync project features.  Ensure the assembly gets sync'd.  If you want to go further, run the assembly pipeline to generate another assembly and ensure it gets sync'd too.

## Related issue
Fixes #123 

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [x] User documentation updated for UI or technical changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Changes
 * [Developer]: Removed deprecated Long constructor.
 * [Developer]: Updated `webpack-dev-server` to fix security vulnerability caused by `websocket-extensions` 0.1.3.
 * [UI]: Update project metadata templates listing page to use Ant Design.
+* [UI]: Allowing users to synchronize assembly data from remote APIs.
```
Comment (deepsidhu85) [inline CHANGELOG.md:11]: Change this to `Added functionality for users to synchronize assembly data from remote APIs.`

File: doc/user/user/project/index.md [hunk 2]
```diff
@@ -304,7 +304,22 @@ Settings for remote synchronized projects can also be managed from the project s
...
 
-IRIDA allows you to synchronize projects between different IRIDA installations.  A remote project appears similar to a local project, but users are not allowed to add samples or sequencing data to a remote project.  Instead all data associated with a remote project will be pulled from a remote IRIDA instance on a regular schedule.  The only data that can be managed for a remote project is the members that are allowed to view the project and associated sample data.
+IRIDA allows you to synchronize projects between different IRIDA installations.  A remote project appears similar to a local project, but users are not allowed to add samples or sequencing data to a remote project.  Instead data associated with a remote project will be updated from a remote IRIDA instance on a regular schedule.  The only data that can be managed for a remote project is the members that are allowed to view the project and associated sample data.
+
+The following data will be synchronized:
+* Basic project metadata
+* FASTQ sequencing data associated with samples
+* Assemblies associated with samples (including both assemblies created by pipeliens within IRIDA, and uploaded assemblies)
```
Comment (deepsidhu85) [inline doc/user/user/project/index.md:312]: Typo `pipeliens`

File: doc/user/user/project/index.md [hunk 3]
```diff
@@ -304,7 +304,22 @@ Settings for remote synchronized projects can also be managed from the project s
...
+* Sample metadata
+
+The following data **will not** be synchronized:
+* Analysis pipeline results
+* Users & groups on a project
+* Analysis automation settings
+* Reference files
+* FAST5 data associated with samples
+
+Some of the above items that are not currently synchronized are expected to be avaliable in future IRIDA versions.
```
Comment (deepsidhu85) [inline doc/user/user/project/index.md:322]: Typo `avaliable`

File: CHANGELOG.md (hunk 1 above)
Comment (tom114) [reply CHANGELOG.md]: Updated in 595a6ad419

File: doc/user/user/project/index.md (hunk 2 above)
Comment (tom114) [reply doc/user/user/project/index.md]: Updated in 595a6ad419

File: doc/user/user/project/index.md (hunk 3 above)
Comment (tom114) [reply doc/user/user/project/index.md]: Updated in 595a6ad419

**Pull request merged**